python/graphstorm/config/argument.py

```diff
@@ -1176,6 +1176,50 @@ def lp_decoder_type(self):
         # Set default value to distmult
         return BUILTIN_LP_DISTMULT_DECODER
 
+    @property
+    def lp_edge_weight_for_loss(self):
+        """ The edge data fields that stores the edge weights used
+            in computing link prediction loss
+
+            The edge_weight can be in following format:
+            1) [weight_name]: global weight name, if an edge has weight,
+            the corresponding weight name is <weight_name>
+            2) ["src0,rel0,dst0:weight0","src0,rel0,dst0:weight1",...]:
```

Comment:
maybe we should use two names. one is lp_edge_weight_per_edge and the other is lp_edge_weight_per_etype.

Reply:
For lp_edge_weight_per_etype do you mean, user only specify the weight of an edge type? Edges with the same edge type use the same weight?
We can support this later. I did not hear similar requirements before.

Reply:
you are dealing with two use cases. I'm thinking if we should treat them separately.

Reply:
We only handle the first use case, i.e., lp_edge_weight_per_edge, in this PR. input option 2 just allows users to use different features (with different feature names) to store edge weights.